src/transport/httpclient.ts

```diff
@@ -0,0 +1,208 @@
+ */
+export class HTTPClientTransport implements Transport {
+  private url: string;
+  private timeout: number;
+  private headers: Record<string, string>;
+  private fetch: typeof fetch;
+  private logger: Logger;
+  private sessionId?: string;
+  private messageHandler?: MessageHandler;
+  private errorCallback?: (error: Error) => void;
+  private closeCallback?: () => void;
+  private isConnected = false;
+
+  constructor(options: HTTPClientTransportOptions) {
+    this.url = options.url;
+    this.timeout = options.timeout || 30000;
+    this.headers = {
+      "Content-Type": "application/json",
+      Accept: "application/json, text/event-stream",
+      ...options.headers,
+    };
+    this.fetch = options.fetch || globalThis.fetch;
+    this.logger = options.logger || createDefaultLogger();
+
+    if (options.enableSessions) {
+      this.logger.debug(
+        "Session support not yet implemented for HTTP client transport"
+      );
+    }
+  }
+
+  async connect(): Promise<void> {
+    try {
+      new URL(this.url);
+      this.isConnected = true;
+      this.logger.info("HTTP Client Transport connected to:", this.url);
+    } catch (error) {
+      const err = new Error(`Invalid URL: ${this.url}`);
+      if (this.errorCallback) this.errorCallback(err);
+      throw err;
+    }
+  }
+
+  async send(message: JSONRPCMessage): Promise<void> {
+    if (!this.isConnected) {
+      throw new Error("Transport not connected. Call connect() first.");
+    }
+
+    let timeoutId: ReturnType<typeof setTimeout> | undefined;
+
+    try {
+      const requestHeaders = { ...this.headers };
+
+      // Add session ID header if available per MCP spec
+      if (this.sessionId) {
+        requestHeaders["Mcp-Session-Id"] = this.sessionId;
+      }
+
+      const controller = new AbortController();
+      timeoutId = setTimeout(() => controller.abort(), this.timeout);
+
+      const response = await this.fetch(this.url, {
+        method: "POST",
+        headers: requestHeaders,
+        body: JSON.stringify(message),
+        signal: controller.signal,
+      });
+
+      if (!response.ok) {
+        throw new Error(`HTTP ${response.status}: ${response.statusText}`);
+      }
+
+      // Check content type to determine response format
+      const contentType = response.headers.get("Content-Type") || "";
+
+      if (contentType.includes("text/event-stream")) {
+        // Handle SSE response
+        await this.handleSSEResponse(response);
+      } else {
+        // Handle JSON response
+        await this.handleJSONResponse(response);
+      }
+    } catch (error) {
+      if (error instanceof Error && error.name === "AbortError") {
+        if (this.errorCallback) this.errorCallback(error);
+        throw new Error(`Request timeout after ${this.timeout}ms`);
+      }
+
+      const unknownError = new Error(`A client error occurred: ${error}`);
+      if (this.errorCallback) this.errorCallback(unknownError);
```

Comment:
nit: prefer explicitly using {} 
```suggestion
      if (this.errorCallback) {
        this.errorCallback(unknownError);
      }
```